File: Beecrowd_Parte1/Assuntos_Pendentes_2984.c
#include <stdio.h>
#include <string.h>

int contarAssuntosPendentes(char *S) {
    int pendentes = 0;
    int balance = 0;
    
    for (int i = 0; S[i] != '\0'; i++) {
        if (S[i] == '(') {
            balance++;
        } else if (S[i] == ')') {
            if (balance > 0) {
                balance--;
            } else {
                pendentes++;
            }
        }
    }
    
    pendentes += balance; // Adiciona os pendentes que não foram fechados
    
    return pendentes;
}

int main() {
    char S[100001];
    fgets(S, sizeof(S), stdin);
    
    // Remover o caractere de nova linha no final da string
    S[strcspn(S, "\n")] = '\0';
    
    int pendentes = contarAssuntosPendentes(S);
    
    if (pendentes > 0) {
        printf("Ainda temos %d assunto(s) pendente(s)!\n", pendentes);
    } else {
        printf("Partiu RU!\n");
    }
    
    return 0;
}

File: Beecrowd_Parte1/Assuntos_Pendentes_2984.java
import java.util.*;

public class Assuntos_Pendentes_2984 {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String S = sc.nextLine();
        sc.close();

        int pendentes = contarAssuntosPendentes(S);
        
        if (pendentes > 0) {
            System.out.printf("Ainda temos %d assunto(s) pendente(s)!\n", pendentes);
        } else {
            System.out.println("Partiu RU!");
        }
    }

    public static int contarAssuntosPendentes(String S) {
        Stack<Character> stack = new Stack<>();
        
        for (int i = 0; i < S.length(); i++) {
            char c = S.charAt(i);
            
            if (c == '(') {
                stack.push(c); // Empilha abertura de parênteses
            } else if (c == ')') {
                if (!stack.isEmpty() && stack.peek() == '(') {
                    stack.pop(); // Fecha um parêntese aberto
                } else {
                    stack.push(c); // Parêntese de fechamento sem abertura correspondente
                }
            }
        }
        
        return stack.size(); // O que sobrar na pilha são os pendentes
    }
}
